-1

Tested at: 73f1d3fb687ca7b3b5d95edc42a778b2289eb5c0

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23158/28037/summary.html

I found follow errors while testing this PR

**Failed tests:  ClangBuild**

* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 16 COMPILER='llvm compile'


 <pre>>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-05-18-1100/src/DQM/Physics/src/EwkElecDQM.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-05-18-1100/src/DQM/Physics/src/TopSingleLeptonDQM.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-05-18-1100/src/DQM/Physics/src/EwkDQM.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-05-18-1100/src/DQM/Physics/src/QcdPhotonsDQM.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-05-18-1100/src/DQM/Physics/src/QcdUeDQM.cc 
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-05-18-1100/src/DQM/Physics/src/FSQDQM.cc:283:17: error: comparison of integers of different signs: 'int' and 'std::vector::size_type' (aka 'unsigned long') [-Werror,-Wsign-compare]
         if(abs(index) < T_trackRec_P4.size()){
            ~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-05-18-1100/src/DQM/Physics/src/TopDiLeptonOfflineDQM.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-05-18-1100/src/DQM/Physics/src/CentralityDQM.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2018-05-18-1100/src/DQM/Physics/src/TopDiLeptonDQM.cc 
</pre>

--- a/DQM/Physics/src/FSQDQM.cc
+++ b/DQM/Physics/src/FSQDQM.cc
@@ -272,7 +272,7 @@ FSQDQM::~FSQDQM()
 	 }
 
 	 float highest_pt_track=-999;
-	 int index=-99;
+	 int index=-999;
 	 for(unsigned int k=0;k<T_trackRec_P4.size();k++){
 	   if(T_trackRec_P4.at(k).Pt() > highest_pt_track){
 		 highest_pt_track=T_trackRec_P4.at(k).Pt();
@@ -280,8 +280,10 @@ FSQDQM::~FSQDQM()
 	       }
 
 	 }
-	 if(abs(index) < T_trackRec_P4.size()){
-	 //	 std::sort(T_trackRec_P4.begin(), T_trackRec_P4.end(), SortByPt());
+	 unsigned int finalid=abs(index);
+	 //	 if(T_trackRec_P4.at(index).Pt()!=0.){
+	 if(finalid < T_trackRec_P4.size()){
+	   //	 std::sort(T_trackRec_P4.begin(), T_trackRec_P4.end(), SortByPt());
 	 for(unsigned int itrk=0;itrk<T_trackRec_P4.size();itrk++){
 	   ++ntracks;
 	     ptsum= ptsum + T_trackRec_P4.at(itrk).Pt();

--- a/DQM/Physics/src/FSQDQM.cc
+++ b/DQM/Physics/src/FSQDQM.cc
@@ -226,7 +226,7 @@ FSQDQM::~FSQDQM()
 	   recoCastorJets.push_back((*castorJets)[ijet]);
 	 }
 	 for (unsigned ijet=0; ijet<recoCastorJets.size(); ijet++) {
-	   cout<<recoCastorJets[ijet].pt()<<endl;
+	   //	   cout<<recoCastorJets[ijet].pt()<<endl;
 	   CastorJetphi->Fill( recoCastorJets[ijet].phi());
 
 	   CastorJetMulti->Fill(recoCastorJets.size());

